asmcomp/cmmgen.ml

```diff
@@ -1346,26 +1355,23 @@ let make_switch arg cases actions dbg =
   let is_const = function
     | Cconst_int n
     | Cconst_pointer n -> (n land 1) = 1
-    | Cconst_natint _
-    | Cconst_float _
+    | Cconst_natint n
+    | Cconst_natpointer n -> (Nativeint.(to_int (logand n one) = 1))
```

Comment:
Allowing the bit to be zero and removing the header sounds fine to me.  Or maybe change the tag to `Abstract_tag` instead of removing the header (might be easier)?